docs/training/apis-howto/api-core-mnist-ug.rst

```diff
@@ -0,0 +1,573 @@
+
+*****************************************************
+ Step 1: Get the Tutorial Files & Run the Experiment
+*****************************************************
+
+To run an experiment, you need, at minimum, a script and an experiment configuration (YAML) file.
+
+Create a new directory.
+
+Get the files from the :download:`core_api_pytorch_mnist.tgz </examples/core_api_pytorch_mnist.tgz>`
+download or from the `Github repository
+<https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api_pytorch_mnist>`_.
+
+.. note::
+
+   Throughout this user guide, we’ll show you how to make modifications to your script. After each
+   step, we’ll re-run our experiment using the appropriate script and its accompanying experiment
+   configuration (YAML) file.
+
+In this initial step, we’ll run our experiment using the ``model_def.py`` script and its
+accompanying ``const.yaml`` experiment configuration file.
+
+CD into the directory and run this command:
+
+.. code::
+
+   det e create const.yaml . -f
+
+Open the Determined WebUI by navigating to ``http://localhost:8080/``. Accept the default determined
+username, leave the password empty, and click **Sign In**.
+
+In the WebUI, select your experiment, and then navigate to the **Logs** tab.
+
+************************
+ Step 2: Report Metrics
+************************
+
+To report training and validation metrics to the Determined master, we’ll add a few lines of code to
+our script. More specifically, we'll create a ``core.Context`` object to allow interaction with the
+master. Then, we'll pass the ``core_context`` as an argument into ``main()``, ``train()``, and
+``test()`` and modify the function headers accordingly.
+
+To run our experiment, we'll use the ``model_def_metrics.py script`` and its accompanying
+``metrics.yaml`` experiment configuration file.
+
+Begin by importing Determined:
+
+.. code::
+
+   import determined as det
+
+.. image:: /assets/images/model-def-metrics-script.png
+
+Step 2.1: Modify the Main Loop
+==============================
+
+#. Modification of __main__ loop to include core_context
+
+Modify the __main__loop to include ``core_context``:
+
+.. note::
+
+   Refer to line 211 in ``model_def_metrics.py``
+
+.. code::
+
+   if __name__ == "__main__":
+
+       # NEW: Establish new determined.core.Context and pass to main function
+       with det.core.init() as core_context:
+           main(core_context=core_context)
+
+Step 2.2: Modify the Train Method
+=================================
+
+Modify the train() method by passing ``core_context`` and adding
+``train.report_training_metrics()``:
+
+Use ``core_context.train`` to report training and validation metrics.
+
+.. note::
+
+   Refer to line 69 in ``model_def_metrics.py``
+
+.. code::
+
+   # NEW: Report training metrics to Determined master via core_context.
+   # Index by (batch_idx + 1) * (epoch-1) * len(train_loader) to continuously plot loss on one graph for consecutive epochs.
```

Comment:
Long lines in code blocks are seriously annoying to the reader.

If you build and view this page, you'll see that the sphinx-produced webpage tries _really_ hard to keep the effective line width of the code snippet constant with all different zooms and window aspect ratios.  That's fantastic if your code snippet is reasonably short because it means that your readers never have to deal with horizontal scroll bars, but if your line length in the code snippet is too long, it means that no matter what, your readers _always_ have to deal with horizontal scroll bars.

Reply:
edited the comments and split them where possible